src/javascripts/Factories/referencedByViewFactory.js

```diff
@@ -9,12 +9,16 @@ ngapp.service('referencedByViewFactory', function(viewFactory) {
         let scope = this.scope;
         scope.tree && factory.releaseTree(scope.tree);
         viewFactory.unlink(this.linkedTreeView, 'linkedReferencedByView');
+        viewFactory.unlink(this.linkedRecordView, 'linkedReferencedByView');
     };
 
     this.linkTo = function(view) {
         if (view.class === 'tree-view') {
```

Comment:
The `referencedByView` should not be able to link to a tree view anymore, if we're linking it to a record view.  a view can't have two links in the same direction, it won't make sense.

Here's a diagram of how view linking should work:

![](http://puu.sh/ynJAe.png)